lib/healthcheck/worker_test.go

```diff
@@ -213,6 +214,66 @@ func Test_dialEndpoints(t *testing.T) {
 	}
 }
 
+func TestGetTargetHealth(t *testing.T) {
+	t.Parallel()
+	ctx := t.Context()
+	enabledHCC := healthCheckConfig{
+		interval:           time.Minute,
+		timeout:            time.Minute,
+		healthyThreshold:   10,
+		unhealthyThreshold: 10,
+	}
+	tests := []struct {
+		desc              string
+		healthCheckConfig *healthCheckConfig
+		dialErr           error
+		wantStatus        types.TargetHealthStatus
+		wantReason        types.TargetHealthTransitionReason
+	}{
+		{
+			desc:              "healthy",
+			healthCheckConfig: &enabledHCC,
+			wantStatus:        "healthy",
+			wantReason:        types.TargetHealthTransitionReasonThreshold,
+		},
+		{
+			desc:       "disabled",
+			wantStatus: "unknown",
+			wantReason: types.TargetHealthTransitionReasonDisabled,
+		},
+		{
+			desc:              "unhealthy",
+			healthCheckConfig: &enabledHCC,
+			dialErr:           trace.Errorf("error dialing"),
+			wantStatus:        "unhealthy",
+			wantReason:        types.TargetHealthTransitionReasonThreshold,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+			worker, err := newWorker(ctx, workerConfig{
+				HealthCheckCfg: test.healthCheckConfig,
+				Target: Target{
+					GetResource: func() types.ResourceWithLabels { return nil },
+					ResolverFn: func(ctx context.Context) ([]string, error) {
+						return []string{"localhost:1234"}, nil
+					},
+					dialFn: func(ctx context.Context, network, addr string) (net.Conn, error) {
+						<-time.After(250 * time.Millisecond)
```

Comment:
have you tried https://go.dev/blog/synctest ? i think we have a few tests doing that already